README.md

```diff
@@ -421,6 +421,17 @@ It is often useful for implementation classes to inherit from each other, if the
 
 However, it is not required! The wrapper classes will have a correct inheritance chain, regardless of the implementation class inheritance chain. Just make sure that, either via inheritance or manual implementation, you implement all of the expected operations and attributes.
 
+### The `[LegacyFactoryFunction]` extended attribute
+
+For interfaces which have the `[LegacyFactoryFunction]` extended attribute, the implementation class file must contain the `legacyFactoryFunction` export, with the signature `(thisValue, globalObject, [...legacyFactoryFunctionArgs], { newTarget, wrapper })`, which is used for:
```

Comment:
I decided to pass the `thisValue` as `arguments[0]`, rather than `this` so that it’s possible to use arrow functions for this:
```js
exports.legacyFactoryFunction = (thisValue, globalObject, [arg0, arg1], { newTarget }) => {
	// code
}
```

I also decided to include the `new.target` and `wrapper` in a `privateData` object to match `constructor(globalObject, args, privateData)`.

Reply:
Would jsdom's implementations actually any of thisValue, newTarget, or wrapper? If not, let's exclude them.

Reply:
We should at the very least include `thisValue` and pass the `args` as an array to leave open the possibility of a `privateData` bag in case <https://html.spec.whatwg.org/multipage/media.html#dom-audio>, <https://html.spec.whatwg.org/multipage/embedded-content.html#dom-image> and <https://html.spec.whatwg.org/multipage/form-elements.html#dom-option> being rewritten.

Reply:
I'm still feeling like this is over-complex given that there are literally three cases and we know exactly how they behave. Our goal here is not to have a general framework to do any possible LegacyFactoryFunction; it's to implement Option, Audio, and Image.

In particular, I'd like to see these either support return override, or use the passed in thisValue, but not both. I suspect thisValue would be simpler, and perhaps necessary to support `class X extends Audio {}`. So I suspect the correct signature here will be `(thisValue, globalObject, [...args])` with no support for return-override.